internal/domain/services/nextDate/nextDate.go

```diff
@@ -0,0 +1,64 @@
+package nextDate
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func NextDate(now time.Time, date string, repeat string) (string, error) {
+	layout := "20060102"
+	var parsedDate time.Time
+	var err error
+	//Если дата не указана, возвращаем текущую
+	if date == "" {
+		parsedDate = now
+	} else {
+		parsedDate, err = time.Parse(layout, date)
+		if err != nil {
+			return "", fmt.Errorf("неправильный формат даты: %v", err)
+		}
+	}
+	//Если правило повторение не указано, возвращаем текущую дату
+	if repeat == "" {
+		if parsedDate.After(now) {
+			return parsedDate.Format(layout), nil
+		}
+		return now.Format(layout), nil
+
+	}
+	//Обработка случая ежегодным повторением
+	if repeat == "y" {
+		parsedDate = parsedDate.AddDate(1, 0, 0)
+		for !parsedDate.After(now) {
+			parsedDate = parsedDate.AddDate(1, 0, 0)
+		}
+		return parsedDate.Format(layout), nil
+	}
+	//Обработка случая с повторением каждые n дней
+	if strings.HasPrefix(repeat, "d ") {
+		repeatDate := strings.Split(repeat, " ")
+		if len(repeatDate) != 2 {
+			return "", fmt.Errorf("неправильный формат правила повторения: %s", repeat)
+		}
+		days, err := strconv.Atoi(repeatDate[1])
+		if err != nil || days <= 0 || days > 400 {
+			return "", fmt.Errorf("неправильное количество дней: %v", err)
```

Comment:
Лайк за оборачивание ошибки, но нужно использовать `%w` (wrap)
```suggestion
			return "", fmt.Errorf("неправильное количество дней: %w", err)
```